x-pack/plugins/task_manager/task.ts

```diff
@@ -53,7 +53,7 @@ export interface RunResult {
    * The state which will be passed to the next run of this task (if this is a
    * recurring task). See the RunContext type definition for more details.
    */
-  state: object;
+  state: Record<string, any>;
```

Comment:
It would force over 12 places to do a type check, as discussed, with the changes required, I'll take a better look at possible approaches in a separate PR.